### DISCUSSION TITLE
pubsub "keyevent@0:expired", redis cluster just return special node connection

### DISCUSSION THREAD

**Discussion by garima803** [redis/go-redis#1835] [Q&A]
As per Redis documentation, Every node of a Redis cluster generates events about its own subset of the keyspace, However, unlike regular Pub/Sub communication in a cluster, events' notifications are not broadcasted to all nodes. keyspace events are node-specific. This means that to receive all keyspace events of a cluster, clients need to subscribe to each of the nodes.
https://redis.io/topics/notifications

**Question:** **Any recommended way to subscribe to all nodes of cluster to receive all keyspace events with  go-redis.**
Or
Is there any way of subscribing dynamically to slave node turn into master in case of fail over (master brought down)?

**Comment by vmihailenco** [comment]
I am no aware of any existing solution, but you can start by describing the "APIs supported by aioredis-cluster" and maybe someone will help.

**Reply by garima803** [reply]
@vmihailenco : Sorry spell mistake, My question was about go-redis. How to subscribe to all nodes of cluster to receive all keyspace events with go-redis.
